File: Braintree/src/test/java/com/braintreepayments/api/models/BraintreeRequestCodesUnitTest.java
package com.braintreepayments.api.models;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static junit.framework.Assert.assertEquals;

@RunWith(JUnit4.class)
public class BraintreeRequestCodesUnitTest {

    @Test
    public void expectedNumberOfRequestCodes() {
        assertEquals(8, BraintreeRequestCodes.class.getDeclaredFields().length);
    }

    @Test
    public void threeDSecure() {
        assertEquals(13487, BraintreeRequestCodes.THREE_D_SECURE);
    }

    @Test
    public void venmo() {
        assertEquals(13488, BraintreeRequestCodes.VENMO);
    }

    @Test
    public void androidPay() {
        assertEquals(13489, BraintreeRequestCodes.ANDROID_PAY);
    }

    @Test
    public void paypal() {
        assertEquals(13591, BraintreeRequestCodes.PAYPAL);
    }

    @Test
    public void visaCheckout() {
        assertEquals(13592, BraintreeRequestCodes.VISA_CHECKOUT);
    }

    @Test
    public void googlePayment() {
        assertEquals(13593, BraintreeRequestCodes.GOOGLE_PAYMENT);
    }

    @Test
    public void ideal() {
        assertEquals(13594, BraintreeRequestCodes.IDEAL);
    }

    @Test
    public void samsungPay() {
        assertEquals(13595, BraintreeRequestCodes.SAMSUNG_PAY);
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/core/BraintreeRequestCodesUnitTest.kt
package com.braintreepayments.api.core

import junit.framework.TestCase.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class BraintreeRequestCodesUnitTest {

    @Test
    fun `THREE_D_SECURE request code is 13487`() {
        assertEquals(13487, BraintreeRequestCodes.THREE_D_SECURE.code)
    }

    @Test
    fun `VENMO request code is 13488`() {
        assertEquals(13488, BraintreeRequestCodes.VENMO.code)
    }

    @Test
    fun `PAYPAL request code is 13591`() {
        assertEquals(13591, BraintreeRequestCodes.PAYPAL.code)
    }

    @Test
    fun `VISA_CHECKOUT request code is 13592`() {
        assertEquals(13592, BraintreeRequestCodes.VISA_CHECKOUT.code)
    }

    @Test
    fun `GOOGLE_PAY request code is 13593`() {
        assertEquals(13593, BraintreeRequestCodes.GOOGLE_PAY.code)
    }

    @Test
    fun `LOCAL_PAYMENT request code is 13596`() {
        assertEquals(13596, BraintreeRequestCodes.LOCAL_PAYMENT.code)
    }

    @Test
    fun `SEPA_DEBIT request code is 13597`() {
        assertEquals(13597, BraintreeRequestCodes.SEPA_DEBIT.code)
    }
}
